.github/workflows/update-plugin-list.yml

```diff
@@ -0,0 +1,32 @@
+name: Update Plugin List
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+jobs:
+  createPullRequest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-verion: 3.8
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install packaging requests tabulate[widechars]
+      - name: Update Plugin List
+        run: python scripts/update-plugin-list.py
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
```

Comment:
Can we do it ourselves with https://github.com/actions/github-script? I prefer to avoid using external actions, especially in little-noticed automatic PRs. If it's difficult, then we should lock it to a specific commit so it doesn't auto-update.

Reply:
Here is the [API reference](https://octokit.github.io/rest.js/v18#pulls-create) which is behind the linked action. The [`README`](https://github.com/actions/github-script) shows, that the action is basically a js script which calls the API. I have no idea how js works and what the environment of the script (provided variables etc.) is, but i am giving it a shot. Seems like i am not getting the `head` right for [this workflow](https://github.com/mcsitter/pytest-plugin-list/runs/1299984596?check_suite_focus=true).

While we are at it, [this](https://github.com/pytest-dev/pytest/pull/7921#discussion_r509431843) discussion is about the target branch of the pull request and could be solved in the same sweep if someone knows how to script this.

Reply:
Your workflow looks right to me. It fails because `head` is `invalid`. I think this is because the branch name `update-plugin-list` doesn't exist in your repo. You should run a previous step which creates it with a `git push --force` command. Also there needs to be some handling of the case where a PR for the branch is already open -- in this case can just reuse it I suppose.

Reply:
I guess this point is still open right?

Reply:
If the github-script or any other roll-our-own solution is too much work, we can keep using `peter-evans/create-pull-request@v3`, but we should at least pin it to a specific commit instead of a tag, so if it's somehow maliciously updated, we won't be affected.